### PR TITLE
Add tests for Open Notebook Folder functionality.

### PR DESCRIPTION
**Pull request by corivera**
Also removed jupyterSettingWriter.ts, since it doesn't get used anywhere.

### CODE REVIEW

File: extensions/notebook/src/test/book/book.test.ts [hunk 1]
```diff
@@ -449,4 +449,110 @@ describe('BookTreeViewProviderTests', function () {
...
+			await Promise.all(bookItems.map(bookItem => bookTreeViewProvider.closeBook(bookItem)));
+		});
+
+		this.afterAll(async function (): Promise<void> {
+			if (await exists(rootFolderPath)) {
+				await promisify(rimraf)(rootFolderPath);
+			}
+		});
+
+		let validateIsBook = (book: BookModel) => {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/test/book/book.test.ts:527]: These seem more generically useful - suggest putting them in a general notebook test helper class

Comment (corivera) [reply extensions/notebook/src/test/book/book.test.ts:527]: The asserts are very specific to this test class. Converting it to a generic method would require a ton of arguments.

Comment (Charles-Gagnon) [reply extensions/notebook/src/test/book/book.test.ts:527]: Which still seems like a good thing to have. Wouldn't it basically be "compare two BookModels to see if they're equivalent"? You shouldn't need a ton of arguments - just the two I would thing.

File: extensions/notebook/src/test/book/book.test.ts [hunk 2]
```diff
@@ -449,4 +449,110 @@ describe('BookTreeViewProviderTests', function () {
...
+		let bookTitle: string;
+		let notebookFolderPath: string;
+		let tableOfContentsFile: string;
+		let standaloneNotebookTitle: string;
+		let standaloneNotebookFile: string;
+		let bookTreeViewProvider: BookTreeViewProvider;
+		let appContext: AppContext;
+
+		this.beforeAll(async () => {
+			rootFolderPath = path.join(os.tmpdir(), `BookFolderTest_${uuid.v4()}`);
```
[the remark this answers is not in the document]
Comment (corivera) [reply extensions/notebook/src/test/book/book.test.ts:465]: I'll try a refactor in a later PR.

**Pull request merged**